### ISSUE TITLE
[macOS] Building the latest Development branch fails

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#14166]
**Describe the bug**
Building `--target Editor` on macOS fails due to the following error:
```
In file included from /Users/macpro13/o3de/osx_build/External/Metal-d17b3d88/Code/CMakeFiles/Atom_RHI_Metal.Private.Static.dir/Unity/unity_0_cxx.cxx:15:
/Users/macpro13/o3de/Gems/Atom/RHI/Metal/Code/Source/RHI/CommandList.cpp:275:21: error: use of undeclared identifier 'srgSlot'
                if (srgSlot == RHI::ShaderResourceGroupData::BindlessSRGFrequencyId && shaderResourceGroup == nullptr)
                    ^
1 error generated.

** BUILD FAILED **


The following build commands failed:
	CompileC /Users/macpro13/o3de/osx_build/External/Metal-d17b3d88/Code/O3DE.build/profile/Atom_RHI_Metal.Private.Static.build/Objects-normal/x86_64/unity_0_cxx.o /Users/macpro13/o3de/osx_build/External/Metal-d17b3d88/Code/CMakeFiles/Atom_RHI_Metal.Private.Static.dir/Unity/unity_0_cxx.cxx normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler 
(in target 'Atom_RHI_Metal.Private.Static' from project 'O3DE')
(1 failure)
```
The issue has been confirmed on the following machines:
- MacBook Air 2020 (M1 chipset, Ventura 13.0.1, Xcode 14.2)
- MacBook Pro 2020 (Intel Core i5, Big Sur 11.3.1, Xcode 13.2.1)

This issue occurs when building both AutomatedTesting and SDK Layout.

Please refer to the attached building logs for more information.

**Steps to reproduce**
1. Create new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
2. Obtain python packages with get_python.sh script located in the o3de/python directory.
3. Create osx_build folder under the o3de location. 
4. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
```
5. Configure and generate the Xcode solution using this command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`
6. Build Mac binaries:
`cmake -E time cmake --build . --target ALL_BUILD --config profile -j 2`

**Expected behavior**
Building completes successfully.

**Actual behavior**
Building fails due to a command failure.

**Found in Branch**
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
[83cf889](https://github.com/o3de/o3de/commit/83cf889897dcd250cd55e35c82b998a262d3d92a)

**Desktop/Device**
- Device: MacBook Pro
- OS: macOS
- Version: Big Sur 11.3.1
- CPU: Intel Core i5 2.0GHz Quad-Core
- GPU: Intel Iris Plus Graphics 1536 MB
- Memory: 16GB

**Additional context**
[macOS_build_fail.txt](https://github.com/o3de/o3de/files/10479266/macOS_build_fail.txt)

**Comment by AMZN-daimini** [comment]
@o3de/sig-graphics-audio FYI

**Comment by moudgils** [comment]
https://github.com/o3de/o3de/pull/14174/files

**Comment by LB-KacperKapusta** [comment]
Issue verified as fixed in Development ([681d577](https://github.com/o3de/o3de/commit/681d577efaef9cc645bf83e3e574296b3b797636)). Closing the issue.
